src/cascadia/TerminalControl/ThrottledFunc.h

```diff
@@ -1,184 +1,55 @@
-    //   is called again before with new arguments, in which case the new
-    //   arguments will be used instead.
-    // - For more information, read the class' documentation.
-    // - This method is always thread-safe. It can be called multiple times on
-    //   different threads.
-    // Arguments:
-    // - args: the arguments to pass to the function
-    // Return Value:
-    // - <none>
     template<typename... MakeArgs>
     void Run(MakeArgs&&... args)
     {
-        if (!_storage.Emplace(std::forward<MakeArgs>(args)...))
-        {
-            _Fire();
-        }
+        _impl(std::forward<MakeArgs>(args)...);
     }
 
-    // Method Description:
-    // - Modifies the pending arguments for the next function invocation, if
-    //   there is one pending currently.
-    // - Let's say that you just called the `Run` method with some arguments.
-    //   After the delay specified in the constructor, the function specified
-    //   in the constructor will be called with these arguments.
-    // - By using this method, you can modify the arguments before the function
-    //   is called.
-    // - You pass a function to this method which will take references to
-    //   the arguments (one argument corresponds to one reference to an
-    //   argument) and will modify them.
-    // - When there is no pending invocation of the function, this method will
-    //   not do anything.
-    // - This method is always thread-safe. It can be called multiple times on
-    //   different threads.
-    // Arguments:
-    // - f: the function to call with references to the arguments
-    // Return Value:
-    // - <none>
     template<typename F>
     void ModifyPending(F f)
     {
-        _storage.ModifyPending(f);
+        _impl.modify_pending(f);
     }
 
 private:
-    winrt::fire_and_forget _Fire()
-    {
-        const auto dispatcher = _dispatcher;
-        auto weakSelf = this->weak_from_this();
-
-        if constexpr (leading)
-        {
-            co_await winrt::resume_foreground(dispatcher);
-
-            if (auto self{ weakSelf.lock() })
-            {
-                self->_func();
-            }
-            else
-            {
-                co_return;
-            }
-
-            co_await winrt::resume_after(_delay);
-
-            if (auto self{ weakSelf.lock() })
-            {
-                self->_storage.Reset();
-            }
-        }
-        else
-        {
-            co_await winrt::resume_after(_delay);
-            co_await winrt::resume_foreground(dispatcher);
-
-            if (auto self{ weakSelf.lock() })
-            {
-                std::apply(self->_func, self->_storage.Extract());
-            }
-        }
-    }
-
     winrt::Windows::UI::Core::CoreDispatcher _dispatcher;
-    winrt::Windows::Foundation::TimeSpan _delay;
-    Func _func;
-
-    ThrottledFuncStorage<Args...> _storage;
+    typename impl::function _func;
+    // As the last member, _throttler will be destroyed first.
```

Comment:
`_throttler`? Do you mean `_impl`?

Reply:
Yep! I had to remove the entire file though. It's important that `til::throttled_func` only resets when the callback is called, but this `ThrottledFunc` delegates the callback to another callback after switching to the main thread. So you technically have to defer the reset until after the switch to the main thread.
Since I was unable to program such a feature I created a full duplicate/copy of `til::throttled_func` and put it in WinRTUtils. 😟

Reply:
Ah well. So it goes.